messenger-app/backend/src/routes/authRoutes.ts

```
@@ -0,0 +1,50 @@
+import express from "express";
+import multer from "multer";
+import { hashPassword, comparePassword } from "../helpers/hashPassword";
+import { createUser, getUserByUsername } from "../services/userService";
+import { generateToken } from "../helpers/jwtFunctions";
+
+const router = express.Router();
+const upload = multer();
+
+router.post("/signup", upload.none(), (req, res) => {
+  getUserByUsername({ username: req.body.username })
+    .then(() => res.status(400).send("Username already exists"))
+    .catch(() => {
+      hashPassword(req.body.password)
```

It is not good to throw an error for an expected and normal case of user not existing. You should throw an error only in case of exceptions: when something wrong is happening. But here, user existence is optional and it is better to use `findUnique` that returns null if the user does not exist